pkg/services/policer/check.go

```diff
@@ -162,6 +166,12 @@ func (p *Policer) processNodes(ctx *processPlacementContext, addr oid.Address,
 		)
 	}
 
+	if typ == object.TypeLock {
+		// all nodes of a container must store the `LOCK` objects
+		// for correct object removal protection
+		shortage = uint32(len(nodes))
```

Comment:
Could you explain it a bit here?
1. We broadcast them on PUT
2. They can't be deleted anyway.

Reply:
> We broadcast them on PUT

yes, but we do not check `LOCK`'s placement after creation (e.g. a node could be turned off for a while when a `LOCK` was put)

> They can't be deleted anyway.

but they would be tried to be deleted on every policer's iteration

the interesting thing about it is that the answers to both of your questions are related: we either try to keep `LOCK`s on every container node or try to remove them unsuccessfully by GC (well, ok, the 3rd option is to change the code significantly)

Reply:
I mean could you add this info to a comment?

Reply:
:hushed:

Reply:
added